x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportForecastJobAction.java

```diff
@@ -124,6 +139,26 @@ private void getForecastRequestStats(String jobId, String forecastId, ActionList
         jobResultsProvider.getForecastRequestStats(jobId, forecastId, forecastRequestStatsHandler, listener::onFailure);
     }
 
+    static Long getAdjustedMemoryLimit(Job job, Long requestedLimit, AbstractAuditor<? extends AbstractAuditMessage> auditor) {
+        if (requestedLimit == null || requestedLimit == DEFAULT_MAX_MODEL_MEMORY.getBytes()) {
```

Comment:
I think we should just make the default `null`.  Doing it the way it's currently done means there's a weird situation where an explicitly configured forecast memory of `20mb` works without capping for a job with a `10mb` `model_memory_limit`, but an explicitly configured forecast memory of `19mb` is capped and generates an audit message.